dashboard/src/components/IasqlEditor.tsx

```diff
@@ -102,11 +101,15 @@ export default function IasqlEditor() {
     const command = {
       name: 'Run SQL',
       bindKey: { win: 'Ctrl-Enter', mac: 'Cmd-Enter' },
-      exec: () => handleQueryToRunUpdate(selectedDb, isRunningSql),
+      exec: () => handleQueryToRunUpdate(selectedDb, editorSelectedTab),
     };
-    editorRef?.current?.editor?.commands?.removeCommand(command);
-    editorRef?.current?.editor?.commands?.addCommand(command);
-  }, [handleQueryToRunUpdate, isRunningSql, selectedDb]);
+    if (editorTabs?.[editorSelectedTab]?.isRunning) {
+      editorRef?.current?.editor?.commands?.removeCommand(command);
+    } else {
+      editorRef?.current?.editor?.commands?.removeCommand(command);
+      editorRef?.current?.editor?.commands?.addCommand(command);
```

Comment:
How often is this branch executed, do you know? Per keystroke?

Reply:
No, either when a new db is selected or when something change in the tabs.